lib/screenplay_web/plugs/ensure_api_auth.ex

```diff
@@ -0,0 +1,19 @@
+defmodule ScreenplayWeb.Plugs.EnsureApiAuth do
+  @moduledoc """
+  Plug used to verify a request contains a valid API key in the x-api-key request header.
+  """
+
+  import Plug.Conn
+
+  def init(default), do: default
+
+  def call(conn, _default) do
+    api_key = Plug.Conn.get_req_header(conn, "x-api-key")
```

Comment:
`get_req_header` returns a list:

```elixir
@spec get_req_header(t(), binary()) :: [binary()]
```

...since some HTTP headers can have multiple values. This won't crash, but also will never allow anything through the plug. We could handle this by matching like `[api_key] = ...`, which should be fine as multiple values being provided for this header would definitely be a "developer error".

Seeing this made me realize there aren't actually any tests for this controller/plug in the PR — if we had one, it would have surfaced this issue right away. Let's add a couple of basic controller tests? (one for getting the `[]` successfully, and one that fails on an invalid API key; we can expand these as we build the real functionality)

Reply:
Yikes, sorry about that.